main_test.go

```diff
@@ -696,6 +697,29 @@ func TestServe(t *testing.T) {
 			},
 			startHTTP,
 		},
+		{
+			"http request with default proxy headers",
```

Comment:
I might miss something but were do you check in this test that the remote addr is modified?

Also, you should add the test that when the proxy is disable, nothing is changed

Reply:
The test would otherwise fill as `allowed_networks: ["10.0.0.0/24"]` is set.

The proxy disabled setting should be covered by the other tests (as the middleware is always used).